hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/action/compact/HoodieCompactor.java

```diff
@@ -182,14 +182,28 @@ public abstract void preCompact(
         .withOperationField(config.allowOperationMetadataField())
         .withPartition(operation.getPartitionPath())
         .build();
-    if (!scanner.iterator().hasNext()) {
-      scanner.close();
-      return new ArrayList<>();
-    }
 
     Option<HoodieBaseFile> oldDataFileOpt =
         operation.getBaseFile(metaClient.getBasePath(), operation.getPartitionPath());
 
+    // Considering following scenario: if all log blocks in this fileSlice is rollback, it returns an empty scanner.
+    // But in this case, we need to give it a base file. Otherwise, it will lose base file in following fileSlice.
+    if (!scanner.iterator().hasNext()) {
+      if (!oldDataFileOpt.isPresent()) {
+        scanner.close();
+        return new ArrayList<>();
+      } else {
+        // TODO: we may directly rename original parquet file if there is not evolution/devolution of schema
```

Comment:
do we know if scanner.getRecords() will return an empty map and not null when scanner.iterator().hasNext() is false. For the fix we have put up, we might end up calling 
```
compactionHandler.handleUpdate(instantTime, operation.getPartitionPath(),
          operation.getFileId(), scanner.getRecords(),
          oldDataFileOpt.get());
```
at L 211. 
when all log files are to be rolledback and only have base file to be compacted.

Reply:
If the file slice only has parquet files, why we still trigger compaction ?

Reply:
> If the file slice only has parquet files, why we still trigger compaction ?

Before we actually do compaction, it is quite difficult to know that new fileSlice only has parquet file. There do have one or more log Files exists which have no valid log blocks in them.

Reply:
yes. it's not feasible to know upfront if all log blocks are invalid while planning compaction.